nova/db/main/legacy_migrations/versions/403_fix_quota_double_spending.py

```diff
@@ -18,20 +18,32 @@
 # http://lists.openstack.org/pipermail/openstack-dev/2013-March/006827.html
 
 
+from sqlalchemy.engine.reflection import Inspector
 from sqlalchemy import MetaData, Table
 from migrate.changeset.constraint import UniqueConstraint
 
 
+COLUMNS = ['project_id', 'user_id', 'resource', 'deleted']
+
+
 def _build_constraint(migrate_engine):
     meta = MetaData()
     meta.bind = migrate_engine
     table = Table('quota_usages', meta, autoload=True)
     return UniqueConstraint(
-        'project_id', 'user_id', 'resource', 'deleted',
+        *COLUMNS,
         table=table,
     )
 
 
 def upgrade(migrate_engine):
+    # check if the constraint already exists
+    inspector = Inspector(migrate_engine)
+    wanted_column_names = tuple(sorted(COLUMNS))
+    for constraint in inspector.get_unique_constraints('quota_usages'):
+        column_names = tuple(sorted(constraint['column_names']))
```

Comment:
Why the `tuple()`? `==` should be safe with lists as well.

Reply:
Wasn't sure about lists, but was sure that tuples will work ;)

Also not sure about the performance of comparing lists, but we only have ~2, so that doesn't matter. Feel like it should be without `tuple`?

Reply:
I've removed the `tuple()`, as we really don't need it and it just adds additional characters to read.

I've also played around with `%timeit` and lists, sets and tuples of this size really don't matter. So the only thing where I would have needed the tuple would have been when I wanted to create a `set()` out of the existing constraints, which I initially did just with names.